frontend/src/fetch_items.test.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import { render, screen, waitFor } from '@testing-library/react';
+import FetchItems from './fetch_items';
+
+const server = setupServer(
+  rest.get('/items', (req, res, ctx) => {
+    return res(ctx.json([{ name: 'AgedBrie', sellIn: 10, quality: 10 }]))
+  })
+)
+
+beforeAll(() => server.listen())
+afterEach(() => server.resetHandlers())
+afterAll(() => server.close())
+
+test('loads and displays the fetched items', async () => {
+  render(<FetchItems/>)
+
+  await screen.findByText(/Aged Brie/i);
+  expect(screen.getByText("Aged Brie")).toBeInTheDocument();
```

Comment:
Will this test fail if we changed the way the we presented the items?

I see this type of thing a lot in React apps where components are layered over and over. This assertion is relying on the behavior of `ItemsIndex` which `FetchItems` is depending on.

You can get away with this when you component tree is small, but as it gets larger, your tests become more and more like integration tests.

Here is an alternative that I like

```jsx
const ItemsListForTesting(props: { items: List<Items>}) => {
  return (
    <ul>
        {items.map(item, index => <li key={index}>{item.name}</li> }
    </ul>
  )
}

test('loads and displays the fetched items', async () => {
  render(<FetchItems component={ItemsListForTesting}>)

  await screen.findByText(/Aged Brie/i);

  expect(screen.getByText("Aged Brie")).toBeInTheDocument();
}
```

The test above is decoupled from the `ItemsIndex` now and both have more freedom to change. Additionally, even though the assertion is the same, it is depending on how we configured the component in the test instead of on a detail of the components implementation.

It is fine the way it is, but I wanted to call this out because it is something I see quite often in client projects, but thos projects have much larger component trees and the pain is proportionate! 😅

Reply:
Good point.  Will adjust.